src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -1152,12 +1117,9 @@ static void CloseRequestResponseSession(RequestResponseAmqpLink requestResponseA
                     {
                         if (error.Condition.Equals(AmqpErrorCode.NotFound))
                         {
-                            if (this.isSessionReceiver)
-                            {
-                                throw new SessionLockLostException(Resources.SessionLockExpiredOnMessageSession);
-                            }
+                            ThrowIfSessionReceiver();
```

Comment:
>ThrowIfSessionReceiver [](start = 28, length = 22)

This name sounds misleading. Rename to something like
`ThrowSessionLostIfSessionReceiver`